addons/source-python/packages/source-python/stringtables/downloads.py

```diff
@@ -85,6 +114,50 @@ def remove_directory(self, directory):
                 # Remove the item from the set
                 self.remove(item)
 
+    def add_from_file(self, file_path):
+        """Add all the paths listed in the file to the downloadables.
+
+        :param str file_path:
+            The file that contains the paths to add to the downloadables.
+            Lines that starts with '#' or '//' will be ignored.
+        :return:
+            Return the number of files that have been added.
+        :rtype: int
+        """
+        file_path = Path(file_path)
+        if not file_path.exists():
+            with_game_path = GAME_PATH.joinpath(file_path)
+            if with_game_path.exists():
```

Comment:
We should not check for the files to be physically present on the server because this is not guaranteed. For example, they could be served from a fast download, only be present as their compressed form (ztmp), or through a custom VPK, etc. Moreover, they might also be resolvable through the `custom/` directory on some games. So yeah, I would just add the files as is to the table without any validation.